atlas/AtlasMl/atlasml/ml/embeddings.py

```
@@ -0,0 +1,141 @@
+            
+        Raises:
+            OpenAIError: If the API request fails.
+        """
+        try:
+            response = self.azure_client.embeddings.create(
+                model="te-3-small",
+                input=description,
+            )
+            return response.data[0].embedding
+        except OpenAIError as e:
+            raise OpenAIError(f"Failed to generate OpenAI embeddings: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)